src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -0,0 +1,496 @@
+            le_sub_iff_add_le, ← ennreal.to_real_add, ennreal.to_real_le_to_real,
+            measure.coe_nnreal_smul, pi.smul_apply, with_density_apply _ (hA.inter hE₁),
+            show ε • μ (A ∩ E) = (ε : ℝ≥0∞) * μ (A ∩ E), by refl,
+            ← set_lintegral_const, ← lintegral_add measurable_const hξm] at this,
+      { rw [ne.def, ennreal.add_eq_top, not_or_distrib],
+        exact ⟨ne_of_lt (measure_lt_top _ _), ne_of_lt (measure_lt_top _ _)⟩ },
+      { exact ne_of_lt (measure_lt_top _ _) },
+      { exact ne_of_lt (measure_lt_top _ _) },
+      { exact ne_of_lt (measure_lt_top _ _) },
+      { rw with_density_apply _ (hA.inter hE₁),
+        exact hξle (A ∩ E) (hA.inter hE₁) },
+      { apply_instance } },
+  -- from this, we can show `ξ + ε * E.indicator` is a function in `measurable_le` with
+  -- integral greater than `ξ`
+    have hξε : ξ + E.indicator (λ _, ε) ∈ measurable_le μ ν,
+    { refine ⟨measurable.add hξm (measurable.indicator measurable_const hE₁), λ A hA, _⟩,
+      have : ∫⁻ a in A, (ξ + E.indicator (λ _, ε)) a ∂μ =
+            ∫⁻ a in A ∩ E, ε + ξ a ∂μ + ∫⁻ a in A ∩ Eᶜ, ξ a ∂μ,
+      { rw [lintegral_add measurable_const hξm, add_assoc,
+            ← lintegral_union (hA.inter hE₁) (hA.inter (hE₁.compl))
+              (disjoint.mono (set.inter_subset_right _ _) (set.inter_subset_right _ _)
+              disjoint_compl_right), set.inter_union_compl],
+        simp_rw [pi.add_apply],
+        rw [lintegral_add hξm (measurable.indicator measurable_const hE₁), add_comm],
+        refine congr_fun (congr_arg has_add.add _) _,
+        rw [set_lintegral_const, lintegral_indicator _ hE₁, set_lintegral_const,
+            measure.restrict_apply hE₁, set.inter_comm] },
+      conv_rhs { rw ← set.inter_union_compl A E },
+      rw [this, measure_union _ (hA.inter hE₁) (hA.inter hE₁.compl)],
+      { exact add_le_add (hε₂ A hA)
+          (hξle (A ∩ Eᶜ) (hA.inter hE₁.compl)) },
+      { exact disjoint.mono (set.inter_subset_right _ _) (set.inter_subset_right _ _)
+          disjoint_compl_right } },
+      have : ∫⁻ a, ξ a + E.indicator (λ _, ε) a ∂μ ≤ Sup (measurable_le_eval μ ν),
+      { exact le_Sup ⟨ξ + E.indicator (λ _, ε), hξε, rfl⟩ },
+  -- but this contradics the maximality of `∫⁻ x, ξ x ∂μ`
+      refine not_lt.2 this _,
+      rw [hξ₁, lintegral_add hξm (measurable.indicator (measurable_const) hE₁),
+          lintegral_indicator _ hE₁, set_lintegral_const],
+      refine ennreal.lt_add_right _ (ennreal.mul_pos.2 ⟨ennreal.coe_pos.2 hε₁, hE₂⟩),
+      have := measure_lt_top (μ.with_density ξ) set.univ,
+      rwa [with_density_apply _ measurable_set.univ, measure.restrict_univ] at this },
+  -- since `μ.with_density ξ ≤ ν`, it is clear that `ν = ν₁ + μ.with_density ξ`
+  { rw hν₁, ext1 A hA,
+    rw [measure.coe_add, pi.add_apply, measure.sub_apply hA hle,
+        add_comm, ennreal.add_sub_cancel_of_le (hle A hA)] },
+end
+
+instance {μ ν : measure α} [finite_measure μ] [finite_measure ν] :
```

Comment:
you shouldn't need finiteness of `nu` here: you always have `singular_part mu nu <= mu`. This should probably go in the preliminaries, around line 88 of this file (and you can already include the same statement for sigma-finite measures). Same thing for the next instance.